File: Javascript/traversy-js-challenges-main/02-basic-challenges-2/04-find-missing-number/find-missing-number.js
function findMissingNumber(arrayOfNumbers) {
    if (arrayOfNumbers.length === 0) {
        return 1;
    }
    const n = arrayOfNumbers.length + 1;
    const expectedSumOfNumbers = (n * (n + 1) / 2);
    let actualSum = 0;
    for (let i=0; i < arrayOfNumbers.length; i++) {
        actualSum += arrayOfNumbers[i]
    }
    return expectedSumOfNumbers - actualSum;

}

module.exports = findMissingNumber;

File: Python/AlgorithmPractice/find_missing_number/find_missing_number.py

class FindMissingNumber:
	def __init__(self, list_of_numbers):
		self.list_of_numbers = list_of_numbers

	def get_missing_number(self):
		if len(self.list_of_numbers) == 1:
			return 1
		n = len(self.list_of_numbers) + 1
		expected_sum_of_numbers = n * (n + 1) / 2
		actual_sum = 0
		for number in self.list_of_numbers:
			actual_sum += number
		return expected_sum_of_numbers - actual_sum

